test/pytest/mockwebserver.py

```diff
@@ -0,0 +1,492 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# https://github.com/astral-sh/ruff/issues/10980#issuecomment-2219615329
+# ruff: noqa: RUF029
+
+import argparse
+import asyncio
+import binascii
+import contextlib
+import json
+import logging
+import os
+import pwd
+import socket
+import weakref
+from collections.abc import AsyncIterator, Awaitable, Callable, Mapping
+from pathlib import Path
+from typing import ClassVar, NamedTuple, Self
+
+import aiohttp
+from aiohttp import WSCloseCode, web
+from yarl import URL
```

Comment:
Hmm what does this URL object offer over `from urllib.parse import urlparse`, looking at the coverage usage  (js_coverage.py), it doesn't seem to hard to switch to urlparse

Reply:
`yarl.URL` is the native URL object of `aiohttp`.  If you're using `aiohttp` then you're already using `yarl`.